airbyte-integrations/connectors/destination-snowflake/src/main/java/io/airbyte/integrations/destination/snowflake/typing_deduping/SnowflakeSqlGenerator.java

```diff
@@ -19,28 +39,396 @@ public ColumnId buildColumnId(String name) {
     return new ColumnId(name, name, name);
   }
 
+  public String toDialectType(final AirbyteType type) {
+    if (type instanceof final AirbyteProtocolType p) {
+      return toDialectType(p);
+    } else if (type instanceof Struct) {
+      // TODO should this+array just be VARIANT?
+      return "OBJECT";
+    } else if (type instanceof Array) {
+      return "ARRAY";
+    } else if (type instanceof UnsupportedOneOf) {
+      return "VARIANT";
+    } else if (type instanceof final Union u) {
+      final AirbyteType typeWithPrecedence = u.chooseType();
+      // typeWithPrecedence is never a Union, so this recursion is safe.
+      return toDialectType(typeWithPrecedence);
+    }
+
+    // Literally impossible; AirbyteType is a sealed interface.
+    throw new IllegalArgumentException("Unsupported AirbyteType: " + type);
+  }
+
+  public String toDialectType(final AirbyteProtocolType airbyteProtocolType) {
+    // TODO verify these types against normalization
+    return switch (airbyteProtocolType) {
+      case STRING -> "VARCHAR";
+      case NUMBER -> "FLOAT";
+      case INTEGER -> "NUMBER(38,0)";
```

Comment:
Note for the futire: `INT` in snowflake is the same as `NUMBER`, which is the same as `NUMBER(38,0)` - https://docs.snowflake.com/en/sql-reference/data-types-numeric#int-integer-bigint-smallint-tinyint-byteint

nit: It might be better to write `BIGINT`, and always pass this alais down to snowflake

Reply:
looks like `NUMBER` is snowflake's internal name (i.e. what you get from querying `information_schema.columns`), using that everywhere :shrug: